lib/absinthe/type/built_ins/spec_compliant_int.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Absinthe.Type.BuiltIns.SpecCompliantInt do
+  use Absinthe.Schema.Notation
+
+  @moduledoc false
+
+  scalar :integer, name: "Int" do
+    description """
+    The `Int` scalar type represents non-fractional signed whole numeric
+    values between `-2^31` and `2^31 - 1`, as outlined in the
+    [GraphQl spec](ttps://spec.graphql.org/October2021/#sec-Int).
+    """
+
+    serialize &__MODULE__.serialize_int/1
+    parse parse_with([Absinthe.Blueprint.Input.Integer], &parse_int/1)
+  end
+
+  @min_int -2 ** 31
+  @max_int 2 ** 31 - 1
+
+  def serialize_int(value) when is_integer(value) and value >= @min_int and value <= @max_int do
+    value
+  end
+
+  defp parse_int(value) when is_integer(value) and value >= @min_int and value <= @max_int do
+    {:ok, value}
+  end
+
+  defp parse_int(_) do
```

Comment:
For followup, when https://github.com/absinthe-graphql/absinthe/pull/1187 is merged a custom error could be returned when the int is out of range.

Reply:
Should this PR wait for this?
Would it just say _the input is out of range (expected an signed integer from XXX to XXX)_?

Reply:
That PR is merged now, let's rebase on master and add a custom error.